bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GCData.java

```diff
@@ -47,6 +48,8 @@ public final class GCData {
 	public float[] lineDashes;
 	public float lineMiterLimit = 10;
 	public int alpha = 0xFF;
+	public int deviceZoom;
+	public int nativeDeviceZoom;
```

Comment:
One rather general question: why do these values have to be stored in the `GCData`? In my understanding, these values can be retrieved from a GC's drawable. So isn't this just some kind of cache? Then if this caching is really necessary (e.g., for performance reasons), is it really necessary to cache the values in `GCData` instead of `GC` if they are only used there? The current implementation also seems to require that a `Control` changes the zoom values in `GCData`, but wouldn't it be better to have a `GC` registered as a listener for zoom change events at the `Control`, so that it can update these information on its own (just like we discussed for any other kind of class that caches some values that need to updated upon zoom changes).